app/scripts/modules/amazon/src/pipeline/stages/deployCloudFormation/evaluateCloudFormationChangeSetExecutionDetails.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+import { IExecutionDetailsSectionProps } from 'core/pipeline';
+import { ExecutionDetailsSection } from '@spinnaker/core';
+import { EvaluateCloudFormationChangeSetExecutionApproval } from './evaluateCloudFormationChangeSetExecutionApproval';
+
+export class EvaluateCloudFormationChangeSetExecutionDetails extends React.Component<IExecutionDetailsSectionProps> {
+  public static title = 'Change Set Execution';
+
+  constructor(props: IExecutionDetailsSectionProps) {
+    super(props);
+  }
+
+  public render() {
+    const { application, execution, stage, current, name } = this.props;
+    const hasReplacement = stage.context.changeSetContainsReplacement;
+    if (hasReplacement && stage.isRunning && stage.context.actionOnReplacement === 'ask') {
```

Comment:
You can abstract this section even more. A lot of the code is repeated, so instead of using `else if` you can return:

```js
const askAction = hasReplacement && stage.isRunning && stage.context.actionOnReplacement === 'ask';
const hasReplacementNotRunning = hasReplacement && !stage.isRunning && stage.context.changeSetExecutionChoice;
const noReplacementNotRunning = !hasReplacement && !stage.isRunning;

<ExecutionDetailsSection name={name} current={current}>
   {askAction ? <EvaluateCloudFormationChangeSetExecutionApproval {...props} /> : 
       <div>
            <div>
              <dl className="no-margin">
                  <dt>ChangeSet Name</dt>
                  <dd>{stage.context.changeSetName}</dd>
                  <dt>Replacement</dt>
                   <dd>{Boolean(hasReplacement)}</dd>
               {stage.context.changeSetExecutionChoice && <div>
                    <dt>Judgment</dt>}
                    <dd>{stage.context.changeSetExecutionChoice}</dd>
                    <dt>Judged By</dt>
                    <dd>{stage.context.lastModifiedBy}</dd>
                </div>}
              </dl>
            </div>
          </div>
   }      
        </ExecutionDetailsSection>
```

If you want to hide `Replacement` field if undefined then you can use a conditional as well

Reply:
Hi @caseyhebebrand  thank you! 

it's committed and travis is running 🤞 
I've changed a bit your proposed logic, to include a message in case there is no changesets. Also had to cast to String in order to be displayed